lib/src/util/windows/user.cc

```diff
@@ -44,15 +46,31 @@ namespace facter { namespace util { namespace windows { namespace user {
 
     string home_dir()
     {
-        string home, alt;
-        if (environment::get("HOME", home)) {
-            return home;
-        } else if (environment::get("HOMEDRIVE", home) && environment::get("HOMEPATH", alt)) {
-            return home + alt;
-        } else if (environment::get("USERPROFILE", home)) {
-            return home;
+        HANDLE temp_token;
+        if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &temp_token)) {
+            LOG_DEBUG("OpenProcessToken call failed: %1%", system_error());
+            return {};
         }
-        return {};
+        scoped_resource<HANDLE> token(move(temp_token), CloseHandle);
```

Comment:
It looks like scoped_resource takes its argument by value and `HANDLE` is just a typedef to a big integer. I don't think we need `move` here?

https://github.com/puppetlabs/facter/blob/master/lib/inc/facter/util/scoped_resource.hpp#L36